### PR TITLE
Release 1.10.0: Staging camel-k-runtime 1.14.0 and kamelet-catalog 0.9.0 tag

### PR DESCRIPTION
**Pull request by oscerd**
Related to #3560

### CODE REVIEW

File: script/Makefile [hunk 1]
```diff
@@ -17,7 +17,7 @@ VERSIONFILE := pkg/util/defaults/defaults.go
 VERSION ?= 1.10.0-SNAPSHOT
```
Comment (squakez) [inline script/Makefile:17]: Maybe this one should be a fixed version `1.10.0`?

**Review by squakez — APPROVED** [review]
One clarification comment. Thanks for taking care!

File: go.mod [hunk 2]
```diff
@@ -64,104 +64,6 @@ require (
 	sigs.k8s.io/controller-runtime v0.10.3
 )
 
-require (
```
Comment (astefanutti) [inline go.mod:67]: I think that's added by newer versions of Go, so may be better to leave that section.

**Review by astefanutti — APPROVED** [review]

File: go.mod (hunk 2 above)
Comment (oscerd) [reply go.mod:67]: I suppose this is because I need to switch to 1.17

Comment (astefanutti) [reply go.mod:67]: Yes  I think so.

**Comment by oscerd** [comment]
Rebased

**Pull request merged**